components/provisioner/internal/director/directorclient.go

```diff
@@ -247,10 +248,34 @@ func (cc *directorClient) executeDirectorGraphQLCall(directorQuery string, tenan
 	req.Header.Set(AuthorizationHeader, fmt.Sprintf("Bearer %s", cc.token.AccessToken))
 	req.Header.Set(TenantHeader, tenant)
 
-	err := cc.gqlClient.Do(req, response)
-	if err != nil {
-		return apperrors.Internal("Failed to execute GraphQL query with Director: %s", err.Error())
+	if err := cc.gqlClient.Do(req, response); err != nil {
+		if egErr, ok := err.(gcli.ExtendedError); ok {
+			return mapDirectorErrorToProvisionerError(egErr).Append("Failed to execute GraphQL request to Director")
+		}
+		return apperrors.Internal("Failed to execute GraphQL request to Director: %v", err)
 	}
 
 	return nil
 }
+
+func mapDirectorErrorToProvisionerError(egErr gcli.ExtendedError) apperrors.AppError {
+	errorCodeValue, present := egErr.Extensions()["error_code"]
+	if !present {
+		return apperrors.Internal("Failed to read the error code from the error response. Original error: %v", egErr)
+	}
+	errorCode, ok := errorCodeValue.(float64)
+	if !ok {
+		return apperrors.Internal("Failed to cast the error code from the error response. Original error: %v", egErr)
+	}
+	switch directorApperrors.ErrorType(errorCode) {
+	case directorApperrors.InternalError, directorApperrors.UnknownError:
+		return apperrors.Internal(egErr.Error())
+	case directorApperrors.Unauthorized, directorApperrors.InsufficientScopes:
+		return apperrors.Forbidden(egErr.Error())
```

Comment:
I am not sure about returning the `Forbidden` status in both cases. It seems like a good fit for `InsufficientScopes` but not sure about the `Unauthorized`.

Reply:
Hm, will the `BadRequest` work better for you? The `Forbidden` one was just my first thought for these two

Reply:
What about some equivalent of 502 or even Internal error? It is not actually user error if Provisioner gets 401/403 from Director but more so some issue with the configuration of Control Plane itself, so I am thinking something like `Bad Gateway` would make the most sense.